src/integration_tests/datasets/e2e_dataset_integration_test.py

```diff
@@ -358,3 +372,99 @@ def test_dataset_without_title(self):
 
             json_response.pop("dataset_id")
             assert dataset_test_data.unit_response.items() == json_response.items()
+
+    def test_dataset_errors(self):
+        """
+        Test that when we upload three datasets with errors, the correct error is published to the error topic.
+        This checks the cloud function works when there are errors in the dataset.
+        There are errors on 3 instances:
+        - When the dataset file extension is not json
+        - When the dataset file is not valid json
+        - When the dataset file is missing required keys
+
+        * We load the sample dataset json files with errors
+        * Upload the dataset files to the dataset bucket
+        * Check the files are not removed from the bucket
+        * Check the error messages are published to the error topic
+
+        """
+
+        session = setup_session()
+        headers = generate_headers()
+
+        # Upload dataset with invalid filename
+        dataset_incorrect_extension = load_json(
+            f"{config.TEST_DATASET_PATH}dataset.json"
+        )
+        dataset_incorrect_extension_filename = create_filename_error_filepath(
+            "integration-test-incorrect-extension"
+        )
+
+        create_dataset_response = create_dataset(
+            dataset_incorrect_extension_filename,
+            dataset_incorrect_extension,
+            session,
+            headers,
+        )
```

Comment:
The `wait_until_dataset_ready` procedure will not be necessary in this case as the process deemed to fail at the validation stage. Recommend the add a flag to skip the procedure.